tests/test-syntax.el

```diff
@@ -54,12 +119,32 @@
   (test-concat-dir "tests/syntax/string.nim")
   '(((33 . 86) . font-lock-string-face)))
 
- (test-faces-by-range
-  "should highlight char of double quote"
+ (test-double-quote-and-next-line
+  "should highlight double quoted string with single quotes"
+  ;; This test makes sure whether highlights after "xxx = " are
+  ;; font-lock-string-face and next lines are not affected by the string
+  ;; face.
   (test-concat-dir "tests/syntax/string.nim")
-  '(((113 . 115) . font-lock-string-face)))
+  '("var endOfQuote = "         "var endOfQuote2 = "
+    "var beginningOfQuote = "   "var beginningOfQuote2 = "
+    "var enclosedQuote = "      "var enclosedQuote2 = "
+    "var escapedSingleQuote = " "var escapedSingleQuote2 = "
+    "var escapedDoubleQuote = " "var escapedDoubleQuote2 = "
+    "var unbalancedDoubleQuote = " "var unbalancedDoubleQuote2 = "
+    "var unbalancedDoubleQuote3 = " "var unbalancedDoubleQuote4 = "
+    "var rawString = r"
+    ))
 
+ ;; Character
+ (test-characters
+  "should highlight characters correctly"
+  (test-concat-dir "tests/syntax/char.nim"))
+
+ ;; Number
  (test-faces-by-range
-  "should highlight comment correctly after char of double quote"
-  (test-concat-dir "tests/syntax/string.nim")
-  '(((119 . 165) . font-lock-comment-face))))
+  "should not highlight numbers by string-face"
```

Comment:
I don't like the idea of `not` tests. Could you just expand it to a few tests for `number-face`?